### DISCUSSION TITLE
Is Binance Beacon Chain suitable for the cosmos namespace?

### DISCUSSION THREAD

**Discussion by GiMa-Maya** [WalletConnect/walletconnect-monorepo#3882] [Developer Support]
Binance Beacon Chain runs on the cosmos sdk. https://github.com/bnb-chain/bnc-cosmos-sdk

Would it be possible to use the cosmos namespace for BNB, or is it better to create a new namespace?

**Comment by bumblefudge** [comment]
Anything that's cosmos SDK should definitely be in the cosmos namechain (same RPC, same addressing, etc-- i.e. the CAIP profile for cosmos should work for all Cosmos SDK chains).  If there are things that truly _work_ differently (say, if CAIP-10s conform to different regexp's or if there is a cosmos-specific way of expressing custom wallet-note RPC methods that varies from chain to chain) tag me in it, but for the most people, you should just use BNB examples when you write a Cosmos namespace :D 

Here's instructions on "profiling" the CAIPs for a given namespace: https://github.com/ChainAgnostic/namespaces/blob/main/CONTRIBUTING.md
There was actually a cosmos namespace with CAIP-2 and CAIP-10 profiled years ago but I suspect it's completely out of date. It might be useful as a starting point but just add your name to the front of the list as lead author if you do an overhaul and don't be afraid to remove parts that no longer apply in 2023:
https://github.com/ChainAgnostic/namespaces/tree/main/cosmos
